composables/use-httpd-base-url.ts

```diff
@@ -0,0 +1,20 @@
+const DEFAULT_HTTPD_BASE_URL = 'http://127.0.0.1:8080'
+
+function useHttpdBaseUrl() {
+  const route = useRoute()
+
+  if (!route.params.node) {
+    return DEFAULT_HTTPD_BASE_URL
+  }
+
+  const [hostname, port] = route.params.node.split(':')
+  if (!hostname || !port) {
+    return DEFAULT_HTTPD_BASE_URL
+  }
+
+  const scheme = port === '443' ? 'https' : 'http'
+
+  return new URL('/api/v1', `${scheme}://${hostname}:${port}`).toString()
```

Comment:
I used the `URL` API here as you had suggested, but I'm not really sure it brings much to the table since I still need to build the base with a template string.

I might be missing something here.

Reply:
What if we first instantiate the URL object, then set its attributes we can't set from the constructor, and _then_ stringify it? :) 

Also, have did you have a look at unjs? What about [ufo](https://unjs.io/packages/ufo)? Always check in unjs for helper libs, they have top notch generic js mini-libs for everything much like VueUse is for front-end stuff (made by the same guys too).

Reply:
RE: Instantiating the URL object first 
I guess that works although it's a bit awkward it needs a valid url in the constructor.

RE: ufo
I generally avoid adding packages if there's a native API alternative. Especially in one-off cases like this.

Ended up using `new URL` and instantiating the properties